### PR TITLE
DROOLS-6726 : Delete BRL column in model view return error

### PR DESCRIPTION
**Pull request by Rikkola**
[DROOLS-6726](https://issues.redhat.com/browse/DROOLS-6726)

[The WAR](https://drive.google.com/file/d/18TMflbYS95KcXrOUsGRma1lu9lKksiFc/view?usp=sharing)

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>
 
* <b>a full downstream build</b> please add comment: <b>Jenkins run fdb</b>
  
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by Rikkola** [comment]
@jomarko ok green build, ready for reviews.

**Comment by Rikkola** [comment]
@jomarko Wrong PR,  hold it! :D

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/model/synchronizers/impl/BRLConditionColumnSynchronizer.java [hunk 1]
```diff
@@ -172,7 +178,63 @@ public void delete(final ColumnMetaData metaData) throws VetoException {
...
+                    index = i;
+                    break;
+                }
+            }
+        }
+
+        if (index >= 0) {
+            factPattern.getConstraintList().removeConstraint(index);
+        }
+    }
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/model/synchronizers/impl/BRLConditionColumnSynchronizer.java:224]:
```suggestion
    private void removePattern(final BRLConditionVariableColumn column,
                               final FactPattern factPattern) {
        final FieldConstraint[] constraints = factPattern.getConstraintList().getConstraints();
        for (int i = 0; i < constraints.length; i++) {
            FieldConstraint constraint = constraints[i];
            if (constraint instanceof CompositeFieldConstraint) {
                final CompositeFieldConstraint compositeFieldConstraint = (CompositeFieldConstraint) constraint;
                final int compositeFieldConstraintIndexByVar = getCompositeFieldConstraintIndexByVar(compositeFieldConstraint,
                                                                                                     column.getVarName());

                if (compositeFieldConstraintIndexByVar >= 0) {
                    compositeFieldConstraint.removeConstraint(compositeFieldConstraintIndexByVar);
                    return;
                }
            } else if (constraint instanceof SingleFieldConstraint) {
                if (((SingleFieldConstraint) constraint).getValue().equals(column.getVarName())) {
                    factPattern.getConstraintList().removeConstraint(i);
                    return;
                }
            }
        }
    }
```

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/model/synchronizers/impl/BRLConditionColumnSynchronizer.java [hunk 2]
```diff
@@ -172,7 +178,63 @@ public void delete(final ColumnMetaData metaData) throws VetoException {
...
+
+    private int getCompositeFieldConstraintIndexByVar(final CompositeFieldConstraint compositeFieldConstraint,
+                                                      final String varName) {
+        int index = -1;
+        for (FieldConstraint fieldConstraint : compositeFieldConstraint.getConstraints()) {
+            if (fieldConstraint instanceof SingleFieldConstraint) {
+                if (((SingleFieldConstraint) fieldConstraint).getValue().equals(varName)) {
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/model/synchronizers/impl/BRLConditionColumnSynchronizer.java:231]:
```suggestion
                if (Objects.equals(((SingleFieldConstraint) fieldConstraint).getValue(), varName)) {
```

**Review by jomarko — COMMENTED** [review]
Some ideas how to simplify a code little bit, could you please have a look?

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/model/synchronizers/impl/BRLConditionColumnSynchronizer.java [hunk 3]
```diff
@@ -172,7 +179,58 @@ public void delete(final ColumnMetaData metaData) throws VetoException {
...
+        int index = -1;
+        for (FieldConstraint fieldConstraint : compositeFieldConstraint.getConstraints()) {
+            if (fieldConstraint instanceof SingleFieldConstraint) {
+                if (Objects.equals(((SingleFieldConstraint) fieldConstraint).getValue(), varName)) {
+                    return index + 1;
+                }
+            }
+            index++;
+        }
+        return index;
```
Comment (jomarko) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/model/synchronizers/impl/BRLConditionColumnSynchronizer.java:233]: Shouldn't return this -1?

**Review by jomarko — COMMENTED** [review]
I tried manually, and it works, however one last question to code.

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/model/synchronizers/impl/BRLConditionColumnSynchronizer.java (hunk 3 above)
Comment (Rikkola) [reply drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-client/src/main/java/org/drools/workbench/screens/guided/dtable/client/widget/table/model/synchronizers/impl/BRLConditionColumnSynchronizer.java:233]: Good point, it likely should.

**Review by jomarko — APPROVED** [review]
Thank you @Rikkola

**Review by RishiRajAnand — APPROVED** [review]

**Comment by Rikkola** [comment]
Jenkins run fdb

**Comment by Rikkola** [comment]
Jenkins run fdb

**Comment by Rikkola** [comment]
@kiegroup/gatekeepers Can I get a merge please.

**Pull request merged**